test/integration/evictions/evictions_test.go

```diff
@@ -161,76 +163,86 @@ func TestConcurrentEvictionRequests(t *testing.T) {
+	var wg sync.WaitGroup
+	// spawn numOfEvictions goroutines to concurrently evict the pods
+	for i := 0; i < numOfEvictions; i++ {
+		wg.Add(1)
 		go func(id int, doneCh chan bool, errCh chan error) {
-			evictionName := fmt.Sprintf(podNameFormat, id)
-			eviction := newEviction(ns.Name, evictionName, deleteOption)
-
+			defer wg.Done()
+			podName := fmt.Sprintf(podNameFormat, id)
+			eviction := newEviction(ns.Name, podName, deleteOption)
 			var e error
-			for {
+			if err := wait.PollImmediate(5*time.Second, 60*time.Second, func() (bool, error) {
 				e = clientSet.Policy().Evictions(ns.Name).Evict(eviction)
-				if errors.IsTooManyRequests(e) {
-					time.Sleep(5 * time.Second)
-				} else {
-					break
+				switch {
+				case errors.IsTooManyRequests(e):
+					return false, nil
+				case errors.IsConflict(e):
+					return false, fmt.Errorf("Unexpected Conflict (409) error caused by failing to handle concurrent PDB updates: %v", e)
+				case e == nil:
+					return true, nil
+				default:
+					return false, e
 				}
-			}
-			if e != nil {
-				if errors.IsConflict(e) {
-					e = fmt.Errorf("Unexpected Conflict (409) error caused by failing to handle concurrent PDB updates: %v", e)
+			}); err != nil {
+				errCh <- err
+			} else {
+				_, err = clientSet.Core().Pods(ns.Name).Get(podName, metav1.GetOptions{})
+				switch {
+				case errors.IsNotFound(err):
+					doneCh <- true
+				case err == nil:
+					errCh <- fmt.Errorf("Pod %q is expected to be evicted", podName)
+					e = clientSet.Core().Pods(ns.Name).Delete(podName, deleteOption)
+					if e != nil {
+						errCh <- e
+					}
+				default:
+					errCh <- err
+					e = clientSet.Core().Pods(ns.Name).Delete(podName, deleteOption)
+					if e != nil {
+						errCh <- e
+					}
 				}
-				errCh <- e
-				return
 			}
-			doneCh <- true
 		}(i, doneCh, errCh)
 	}
 
-	doneCount := 0
-loop:
-	for {
-		select {
-		case err := <-errCh:
-			t.Errorf("%v", err)
-			break loop
-		case <-doneCh:
-			doneCount++
-			if doneCount == 10 {
-				break loop
-			}
-		case <-time.After(defaultTimeout):
-			t.Errorf("Eviction did not complete within %v", defaultTimeout)
-			break loop
-		}
+	wg.Wait()
+
+	close(doneCh)
+	close(errCh)
+	var errList []error
+	if err := clientSet.Policy().PodDisruptionBudgets(ns.Name).Delete(pdb.Name, deleteOption); err != nil {
+		errList = append(errList, fmt.Errorf("Failed to delete PodDisruptionBudget: %v", err))
+	}
+	for err := range errCh {
+		errList = append(errList, err)
+	}
+	if len(errList) > 0 {
+		t.Fatal(utilerrors.NewAggregate(errList))
 	}
 
-	for i := 0; i < 10; i++ {
-		podName := fmt.Sprintf(podNameFormat, i)
-		_, err := clientSet.Core().Pods(ns.Name).Get(podName, metav1.GetOptions{})
-		if !errors.IsNotFound(err) {
-			t.Errorf("Pod %q is expected to be evicted", podName)
-			if err == nil {
-				clientSet.Core().Pods(ns.Name).Delete(podName, deleteOption)
-			}
+	doneCount := 0
+	for done := range doneCh {
+		if done {
```

Comment:
it doesn't look like we ever send `false` over `doneCh` so it's probably fine to increment immediately without the check. Also we _could_ make `doneCh` of type struct since we don't do anything with the value going over the channel. Since we adopted a shared variable pattern already (with the sync.WaitGroup) you could have a shared counter that you update via `atomic.AddUint32` or similar so this could be cleaned up slightly since you could simply check the value of the counter at the end without ranging over the channel that doesn't provide us a lot of useful information